Spigot-Server-Patches/0218-Add-UnknownCommandEvent.patch

```diff
@@ -0,0 +1,34 @@
+From 399a532756ce4da82c29e07fb1e65d2bf526bbbb Mon Sep 17 00:00:00 2001
+From: Sweepyoface <github@sweepy.pw>
+Date: Wed, 14 Jun 2017 11:12:10 -0400
+Subject: [PATCH] Add UnknownCommandEvent
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index df6a75b0..a73999d0 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -78,6 +78,7 @@ import org.bukkit.craftbukkit.util.Versioning;
+ import org.bukkit.craftbukkit.util.permissions.CraftDefaultPermissions;
+ import org.bukkit.entity.Entity;
+ import org.bukkit.entity.Player;
++import org.bukkit.event.command.UnknownCommandEvent;
+ import org.bukkit.event.inventory.InventoryType;
+ import org.bukkit.event.player.PlayerChatTabCompleteEvent;
+ import org.bukkit.event.world.WorldInitEvent;
+@@ -671,7 +672,11 @@ public final class CraftServer implements Server {
+ 
+         // Spigot start
+         if (StringUtils.isNotEmpty(org.spigotmc.SpigotConfig.unknownCommandMessage)) {
+-            sender.sendMessage(org.spigotmc.SpigotConfig.unknownCommandMessage);
++            // Paper start
++            UnknownCommandEvent event = new UnknownCommandEvent(sender, commandLine, org.spigotmc.SpigotConfig.unknownCommandMessage);
++            Bukkit.getServer().getPluginManager().callEvent(event);
```

Comment:
This needs to check if the message is null and only call send message if its not null.

This will let plugins ignore sending a message if they want to automatically route it to another command, or for example send the message by action bar.